xbmc/windowing/gbm/GBMUtils.cpp

```diff
@@ -103,3 +105,79 @@ void CGBMUtils::ReleaseBuffer()
   m_bo = m_next_bo;
   m_next_bo = nullptr;
 }
+
+// --- CGBMBufferObject ----------------------------------------------------
+
+#if defined (HAS_GBM_BO_MAP)
+
+CGBMBufferObject::CGBMBufferObject(int format) :
+  m_format(format)
+{
+  m_device = dynamic_cast<CWinSystemGbmGLESContext*>(CServiceBroker::GetWinSystem())->GetGBMDevice();
```

Comment:
I prefer `static_cast` here. If a GBM class isn't returned in a GBM-only file, that's a programming error and we should crash, so there's no need to hit RTTI (the extra data c++ uses to keep track of class hierarchies).

If we do use `dynamic_cast`, this signifies an architecture where NULL could possibly result, so we always need a null check after dynamic_cast.